examples/layout_padded/layout_padded.cpp

```diff
@@ -0,0 +1,648 @@
+    for(std::size_t r = 0; r < e.rank(); ++r) {
+      product_of_extents *= std::size_t(e.extent(r));
+    }
+    const size_t min_num_elements_storage = mapping.required_span_size();
+    assert(min_num_elements_storage >= product_of_extents);
+    if(storage.size() < min_num_elements_storage) {
+      storage.resize(min_num_elements_storage);
+    }
+  }
+
+  mdspan_type x(storage.data(), mapping);
+
+  static_assert(std::is_same<mapping_type, typename layout_type::template mapping<extents_type>>::value,
+		"layout_padded::mapping type wrong");
+  static_assert(not mapping_type::is_always_exhaustive());
+  static_assert(mapping_type::is_always_strided());
+  static_assert(mapping_type::is_always_unique());
+  assert(mapping.is_exhaustive() == expect_exhaustive);
+  assert(mapping.is_strided());
+  assert(mapping.is_unique());
+
+  assert(e == x.extents());
+  assert(e == x.mapping().extents());
+
+  using expected_inner_layout_type = details::storage_order_to_layout_t<order>;
+  using expected_inner_mapping_type =
+    typename expected_inner_layout_type::template mapping<expected_inner_extents_type>;
+  expected_inner_mapping_type expected_inner_mapping{expected_inner_extents};
+
+  for (std::size_t r = 0; r < x.rank(); ++r) {
+    assert(x.stride(r) == expected_inner_mapping.stride(r));
+    assert(x.stride(r) == details::ceiling_power_of_two(x.stride(r)));
+  }
+}
+
+void test_layout_padded()
+{
+  // storage will get resized as needed in the tests.
+  std::vector<float> storage;
+
+  constexpr std::integral_constant<StorageOrder, StorageOrder::LEFT> left{};
+  constexpr std::integral_constant<StorageOrder, StorageOrder::RIGHT> right{};
+
+  {
+    constexpr std::integral_constant<std::size_t, 4> overalignment_factor{};
+    using extents_type = stdex::extents<int, 3, 5>;
+    extents_type e{};
+    {
+      using expected_inner_extents_type = stdex::extents<int, 4, 5>;
+      expected_inner_extents_type expected_inner_extents{};
+      const bool expect_exhaustive = false;
+      test_layout_padded_one(storage, left, overalignment_factor, e, expected_inner_extents, expect_exhaustive);
+    }
+    {
+      using expected_inner_extents_type = stdex::extents<int, 3, 8>;
+      expected_inner_extents_type expected_inner_extents{};
+      const bool expect_exhaustive = false;
+      test_layout_padded_one(storage, right, overalignment_factor, e, expected_inner_extents, expect_exhaustive);
+    }
+  }
+}
+
+int main()
+{
+  static_assert(details::my_bit_width(0) == 0);
+  static_assert(details::my_bit_width(1) == 1);
+  static_assert(details::my_bit_width(2) == 2);
+  static_assert(details::my_bit_width(5) == 3);
+  static_assert(details::my_bit_width(7) == 3);
+  static_assert(details::my_bit_width(8) == 4);
+
+  static_assert(details::ceiling_power_of_two(3) == std::size_t(4));
+  static_assert(details::ceiling_power_of_two(2) == std::size_t(2));
+  static_assert(details::ceiling_power_of_two(1) == std::size_t(1));
```

Comment:
Move these closer to their definition

Reply:
@dalg24 I ended up not needing these for the example, based on Christian's feedback.